js/ProductCat.js

```diff
@@ -170,6 +171,7 @@
   const sendGetRequest = (path, params, options, callback, failure) => {
     const endpoint = params.country == "CN" ? CN_SERVER_URL : SERVER_URL;
     params.cid =settings.cid;
+    params.uid =settings.uid || generateUUID();
```

Comment:
if settings.uid is null, a new UID is sent everytime. Should store the new UID into localStorage.
Suggest to create a function for this logic, since it is the same for both GET and POST.

Reply:
in browser environment if uid is not setup it will set in localstorage in the early step .

here is in case in node environment it will generate a new one ...

Reply:
can we make it persistent then? i saw the code below to store into local storage, but what if it fail for whatever reason? Since we are providing Node version, we need to be consistent. The UID should not change for every request.